File: anko/src/main/java/com/example/anko/Main3Activity.java
package com.example.anko;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

public class Main3Activity extends AppCompatActivity {
        LoopScaleView mLsv4;
        TextView mTvValue4;

        @Override
        protected void onCreate(Bundle savedInstanceState) {
                super.onCreate(savedInstanceState);
                setContentView(R.layout.activity_main3);
                initView();
                mLsv4.setOnValueChangeListener(new ValueChangeListener(4));
        }

        public void initView() {

                mLsv4 = findViewById(R.id.lsv_4);
                mTvValue4 = findViewById(R.id.tv_value4);
        }

        class ValueChangeListener implements LoopScaleView.OnValueChangeListener {

                private int type;

                public ValueChangeListener(int type) {
                        this.type = type;
                }

                @Override
                public void OnValueChange(int newValue) {
                        switch (type) {
                                case 1:

                                        break;
                                case 2:

                                        break;
                                case 3:

                                        break;
                                case 4:
                                        mTvValue4.setText("长度 " + newValue + " cm");
                                        break;
                        }
                }
        }
}

File: app/src/main/java/com/canbot/stdlibkt/mvvm/Main3Activity.kt
package com.canbot.stdlibkt.mvvm

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.widget.Toast
import com.canbot.stdlibkt.R
import com.canbot.stdlibkt.databinding.ActivityMain3Binding

class Main3Activity : AppCompatActivity() ,RepositoryRecyclerViewAdapter.OnItemClickListener {

        lateinit var binding: ActivityMain3Binding
        private val repositoryRecyclerViewAdapter = RepositoryRecyclerViewAdapter(arrayListOf(), this)
        override fun onCreate(savedInstanceState: Bundle?) {
                super.onCreate(savedInstanceState)
                //setContentView(R.layout.activity_main3)
                binding = DataBindingUtil.setContentView(this, R.layout.activity_main3)
                var viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
                binding.viewModel = viewModel
                binding.executePendingBindings()

                binding.repositoryRv.layoutManager = LinearLayoutManager(this)
                binding.repositoryRv.adapter = repositoryRecyclerViewAdapter
                viewModel.repositories.observe(this, Observer { it?.let { repositoryRecyclerViewAdapter.replaceData(it) } })

        }

        override fun onItemClick(position: Int) {
                Toast.makeText(this, "您点击的是$position", Toast.LENGTH_SHORT).show()
        }
}
